File: android/app/src/main/java/com/candy/handyman/data/remote/dto/NotificationDto.kt
package com.candy.handyman.data.remote.dto

data class NotificationDto(
    val id: String,
    val title: String,
    val content: String,
    val type: String,
    val isRead: Boolean,
    val relatedId: String?,
    val relatedType: String?,
    val image: String?,
    val createdAt: String
)

data class PagedNotificationResult(
    val items: List<NotificationDto>,
    val total: Int,
    val page: Int,
    val pageSize: Int
)

data class NotificationSettingDto(
    val type: String,
    val enabled: Boolean
)

data class UnreadCountDto(
    val count: Int
)

File: src/CandyHandyman.Application/DTOs/NotificationDto.cs
using CandyHandyman.Core.Enums;

namespace CandyHandyman.Application.DTOs;

public class NotificationDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public NotificationType Type { get; set; }
    public bool IsRead { get; set; }
    public Guid? RelatedId { get; set; }
    public string? RelatedType { get; set; }
    public string? Image { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class NotificationSettingDto
{
    public NotificationType Type { get; set; }
    public bool Enabled { get; set; }
}

public class PagedResult<T>
{
    public List<T> Items { get; set; } = new();
    public int Total { get; set; }
    public int Page { get; set; }
    public int PageSize { get; set; }
}
